fhir-server-test/src/test/java/com/ibm/fhir/server/test/ReadTest.java

```diff
@@ -0,0 +1,89 @@
+
+    @Test(groups = { "server-read" })
+    public void testCreatePatient() throws Exception {
+        WebTarget target = getWebTarget();
+
+        // Build a new Patient and then call the 'create' API.
+        Patient patient = TestUtil.readLocalResource("Patient_JohnDoe.json");
+
+        patient = patient.toBuilder().gender(AdministrativeGender.MALE).build();
+        Entity<Patient> entity =
+                Entity.entity(patient, FHIRMediaType.APPLICATION_FHIR_JSON);
+        Response response =
+                target.path("Patient").request()
+                .post(entity, Response.class);
+        assertResponse(response, Response.Status.CREATED.getStatusCode());
+
+        // Get the patient's logical id value.
+        patientId = getLocationLogicalId(response);
+
+        // Next, call the 'read' API to retrieve the new patient and verify it.
+        response = target.path("Patient/"
+                + patientId).request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Patient responsePatient = response.readEntity(Patient.class);
+        TestUtil.assertResourceEquals(patient, responsePatient);
+    }
+
+    @Test(groups = { "server-read" }, dependsOnMethods = {"testCreatePatient" })
+    public void testReadPatient_elements() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient/" + patientId)
+                .queryParam("_elements", "active")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Patient patient = response.readEntity(Patient.class);
+        assertNotNull(patient);
+    }
+
+    @Test(groups = { "server-read" }, dependsOnMethods = {"testCreatePatient" })
+    public void testReadPatient_summary() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient/" + patientId)
+                .queryParam("_summary", "true")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Patient patient = response.readEntity(Patient.class);
+        assertNotNull(patient);
+    }
+
+    @Test(groups = { "server-read" }, dependsOnMethods = {"testCreatePatient" })
+    public void testReadPatient_nonGeneralSearchParameter() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient/" + patientId)
+                .queryParam("_lastUpdated", "ge2020-11-02")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+            "Read only supports general search parameters.");
+    }
+
```

Comment:
maybe add a test where you specify both _elements and _summary and verify _elements gets processed.

Reply:
Added test.